lib/smbios.c

```diff
@@ -572,10 +621,36 @@ ulong write_smbios_table(ulong addr)
 
 	ctx.node = ofnode_null();
 	if (IS_ENABLED(CONFIG_OF_CONTROL)) {
+		int ret;
+
 		uclass_first_device(UCLASS_SYSINFO, &ctx.dev);
-		if (ctx.dev)
+		if (ctx.dev) {
+			printf("=====================> ctx.dev->name: %s\n", ctx.dev->name);
+
 			parent_node = dev_read_subnode(ctx.dev, "smbios");
+		} else {
+			/* if no device found, bind the platform-specific driver */
+			printf("=====================> no ctx.dev 1\n");
+
+			ret = device_bind_driver(NULL, "sysinfo_smbios_plat", "sysinfo_smbios_plat_drv", &ctx.dev);
```

Comment:
I don't think you need this. 
If driver is present which is what you find based on u-boot,sysinfo-smbios compatible string it will be present. 
Of different driver will be present.
If none of them is there you will just fill default values. And this should be behavior even today without u-boot,sysinfo-smbios node. 
I know that you are taking some information from DT even in these default case but I don't think it is needed to be supported. You can use generic/default values as is done 
t->bios_release_date = smbios_add_prop(ctx, NULL, U_BOOT_DMI_DATE);
t->vendor = smbios_add_prop(ctx, NULL, "U-Boot");
etc.

As you see here if model or compatible string is not filled (which is maybe possible too) you get NULL string and default passed value is used.
                if (ofnode_valid(ctx->node)) {
                        str = ofnode_read_string(ctx->node, prop);
                } else {
                        const struct map_sysinfo *nprop;

                        nprop = convert_sysinfo_to_dt(ctx->subnode_name, prop);
                        get_str_from_dt(nprop, str_dt, sizeof(str_dt));
                        str = (const char *)str_dt;
                }

                ret = smbios_add_string(ctx, str && *str ? str : dval);

Reply:
@michalsimek just be clear here 

We could define the arm cache driver in the arm architecture folders and then 'load it' with 

+U_BOOT_DRIVER(sysinfo_generic_arm) = {
 .name           = "sysinfo_arm",
  .id             = UCLASS_SYSINFO,
 .ops            = &xxxxx_ops,
 .flags          = DM_FLAG_PRE_RELOC,
}

This is partially done in drivers/sysinfo/smbios_plat.c already, but I think we are better of moving it to arm specific folders

Reply:
@apalos
The arm64 specific code is in arch/arm/cpu/armv8/sysinfo.c.
However we still need drivers/sysinfo/smbios_plat.c on top of it with weak functions for those platforms other than arm64.

@michalsimek 
So far as I under stand, `smbios_add_prop` and `smbios_add_prop_si` are just for strings, but for other types for example int we still need to call `sysinfo_get_int` - I didn't find an smbios API can retrieve int. Maybe I am missing something.